app/src/main/java/org/ea/sqrl/activites/EnableQuickPassActivity.java

```diff
@@ -0,0 +1,96 @@
+package org.ea.sqrl.activites;
+
+import android.os.Bundle;
+import android.view.Gravity;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.PopupWindow;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import org.ea.sqrl.R;
+import org.ea.sqrl.activites.base.LoginBaseActivity;
+import org.ea.sqrl.database.IdentityDBHelper;
+import org.ea.sqrl.processors.SQRLStorage;
+import org.ea.sqrl.utils.SqrlApplication;
+
+import static android.view.View.GONE;
+
+public class EnableQuickPassActivity extends LoginBaseActivity {
+    String TAG = "EnableQuickPassActivity";
+    boolean mPopupShown = false;
+    boolean mDoingLogin = false;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+    }
+
+    public void doingLogin() {
+        mDoingLogin = true;
+    }
+
+    public void failedLogin() {
+        mDoingLogin = false;
+    }
+
+    @Override
+    public void onWindowFocusChanged(boolean hasFocus) {
+        super.onWindowFocusChanged(hasFocus);
+        if (hasFocus) {
+            if (!mPopupShown) {
+                SQRLStorage storage = SQRLStorage.getInstance(EnableQuickPassActivity.this.getApplicationContext());
+                String identityName = IdentityDBHelper.getInstance(this).getIdentityName(SqrlApplication.getCurrentId(this));
+                if ("".equals(identityName)) {
```

Comment:
The current function will not return an empty string if the user is missing. I think we will throw an exception. Maybe we should handle this better and return null so we can handle this edge case.

Reply:
When I coded it, I was thinking `getIdentityName()` would not return null (I should have checked).  Now I see that it's pulling using `Map.get(currentId)`, so if current ID is default and the map has no entry, it's going to be null.  So I guess this could be changed to `if (identityName == null) ... `

Back to a point of how best to manage this change in the source control system, should it be added to this branch (that's already been merged), or is it better to put it in another branch?

Reply:
You can do more in this branch if you like and create a new pull request against it. Or create a new branch and then make a pull request against the new branch.

Either way, you need to rebase and keep the branch up to date with master.

Reply:
That's good.  I understand the *goal*, which is to make this branch the same as `origin/master` (plus my two little edits), do a push and do another PR.

The problem is I've never been able to use the <b>rebase</b> function in Android Studio.  My alternative has been to merge from `origin/master`, then push and PR.  That's what I did just now with 5fa23b7 / df26635 , and for some reason the push took a really long time, like 5 minutes!

As to the rebase process, I'm pretty sure I'm not setting the parameters correctly.  I don't know what i need to set to current and to selected to before I select "rebase current onto selected".

I tried many variations, but here's one of my many non-working examples ;)

I checked out `master0609` from `origin/master`, then I checked out `app-shortcuts` (one commit ahead and seven behind `origin/master`).  When I compare, I see the expected differences.  Then I select `VCS > GIT > Rebase...`, complete the dialog as follows:

![rebaseattempt](https://user-images.githubusercontent.com/15800701/59165577-04dcb980-8aec-11e9-9882-b1788b29c2e5.jpg)

> 7:21 PM	Rebase Failed
				Rebasing (1/1)
				interactive rebase in progress; onto 5fa23b7
				Last command done (1 command done):
				pick 5fa23b7 suggested fixes
				No commands remaining.
				You are currently rebasing branch 'app_shortcuts' on '5fa23b7'.
				nothing to commit, working tree clean
				The previous cherry-pick is now empty, possibly due to conflict resolution.
				If you wish to commit it anyway, use:
				git commit --allow-empty
				Otherwise, please use 'git reset'
				Could not apply 5fa23b7... suggested fixes

Reply:
Thanks @sesam.  Your technique avoids the rebase function and also the branch merge function.  I've done the same thing before, but instead of the cherry pick, I just copied the file into a different place in the file system then checked-out origin/master, then manually pasted my changes.  I like the idea of cherry pick, but I've not used that (yet).

In the case, above, I had already made the change to the app-shortcuts, committed and pushed.  But the branch was seven commits behind origin/master since Daniel had done a bunch of merging.

Anyway, thanks for sharing your source control technique!

Reply:
Another way if you want to use something built in to move files away, then you can stash your files and then pop them back into another branch. Usually, when the changes are small it will work.